client/extensions/woocommerce/state/ui/product-categories/selectors.js

```diff
@@ -0,0 +1,88 @@
+ */
+export function getAllProductCategoryEdits( rootState, siteId = getSelectedSiteId( rootState ) ) {
+	return get( rootState, [ 'extensions', 'woocommerce', 'ui', 'productCategories', siteId, 'edits' ], {} );
+}
+
+/**
+ * Gets the accumulated edits for a category, if any.
+ *
+ * @param {Object} rootState Global state tree
+ * @param {Number|Object} categoryId The id of the product category (or { index: # } if pending create)
+ * @param {Number} [siteId] Site ID to check. If not provided, will use selected Site ID
+ * @return {Object} The current accumulated edits
+ */
+export function getProductCategoryEdits( rootState, categoryId, siteId ) {
+	const edits = getAllProductCategoryEdits( rootState, siteId );
+	const bucket = isNumber( categoryId ) && 'updates' || 'creates';
+	const array = get( edits, bucket, [] );
+
+	return find( array, ( c ) => categoryId === c.id );
+}
+
+/**
+ * Gets a category with local edits overlayed on top of fetched data.
+ *
+ * @param {Object} rootState Global state tree
+ * @param {Number|Object} categoryId The id of the product category (or { index: # } if pending create)
+ * @param {Number} [siteId] Site ID to check. If not provided, will use selected Site ID
+ * @return {Object} The category data merged between the fetched data and edits
+ */
+export function getProductCategoryWithLocalEdits( rootState, categoryId, siteId ) {
+	const existing = isNumber( categoryId );
+
+	const category = existing && getProductCategory( rootState, categoryId, siteId );
+	const categoryEdits = getProductCategoryEdits( rootState, categoryId, siteId );
+
+	return ( category || categoryEdits ) && { ...category, ...categoryEdits } || undefined;
+}
+
+/**
+ * Gets all categories, either fetched, edited, or both.
+ *
+ * This returns a list of all fetched categories overlaid with updates (if any) and
+ * all categories in the creates list as well.
+ *
+ * @param {Object} rootState Global state tree
+ * @param {Number} [siteId] Site ID to check. If not provided, will use selected Site ID
+ * @return {Object} The category list merged between the fetched data, edits, and creates
+ */
+export function getProductCategoriesWithLocalEdits( rootState, siteId ) {
+	const categoryCreates = getAllProductCategoryEdits( rootState, siteId ).creates || [];
+	const fetchedCategoriesWithUpdates = getProductCategories( rootState, siteId ).map(
+		( c ) => getProductCategoryWithLocalEdits( rootState, c.id, siteId )
+	);
+
+	return [ ...categoryCreates, ...fetchedCategoriesWithUpdates ];
+}
+
+/**
+ * Gets the product category last edited in the UI.
+ *
+ * @param {Object} rootState Global state tree
+ * @param {Number} [siteId] Site ID to check. If not provided, will use selected Site ID
+ * @return {Object} The category data merged between the fetched data and edits
+ */
+export function getCurrentlyEditingProductCategory( rootState, siteId ) {
+	const edits = getAllProductCategoryEdits( rootState, siteId );
```

Comment:
Yeah, I could, and I don't think it would reduce the readability of the code. Thanks!